src/backend/distributed/executor/intermediate_results.c

```diff
@@ -596,26 +596,33 @@ CreateIntermediateResultsDirectory(void)
 {
 	char *resultDirectory = IntermediateResultsDirectory();
 
-	if (!CreatedResultsDirectory)
+	MemoryContext intermediateResultsContext = AllocSetContextCreateExtended(
+		TopTransactionContext,
+		"Intermediate Result Files Context",
+		ALLOCSET_DEFAULT_MINSIZE,
+		ALLOCSET_DEFAULT_INITSIZE,
+		ALLOCSET_DEFAULT_MAXSIZE);
+
+	MemoryContext oldContext = MemoryContextSwitchTo(intermediateResultsContext);
+
+	int makeOK = mkdir(resultDirectory, S_IRWXU);
+	if (makeOK != 0)
 	{
-		int makeOK = mkdir(resultDirectory, S_IRWXU);
-		if (makeOK != 0)
+		/* if it already exists, someone else beat us to it, that's ok */
+		if (errno != EEXIST)
 		{
```

Comment:
why are we removing the return here? wouldn't that mean multiple processes try to clean up the same directory?